Lab 4

KseniyaKazachenok opened the pull request:
Динамическое программирование

KseniyaKazachenok committed:
Lab 4

Динамическое программирование

EgorOb commented on Автомобиль. Количество маршрутов.py:
```diff
@@ -0,0 +1,39 @@
+    if n <= 0 or m <= 0:
+        raise ValueError
+
+    table = [[0] * m] * n
```
По коду вижу, что вы алгоритм написали верно, но инициализация двумерной матрицы неправильная, так как в таком случае позиционные элементы в каждой строке будут связаны и будут с одинаковым значением. Такую проблему поднимали в группе в телеграмме https://stackoverflow.com/questions/2397141/how-to-initialize-a-two-dimensional-array-in-python
Ну и можно было бы сразу инициализировать двумерную матрицу единицами
```python
[[1] * m for _ in range(n)]
```

EgorOb reviewed (approved):
Зачтено

KseniyaKazachenok committed:
Update Автомобиль. Количество маршрутов.py

Поправила создание таблицы
Убрала лишнее условие про n или m == 0
Поправила range
